configure-places-in-the-sdk/places-extension/using-places-without-active-region-monitoring.md

```diff
@@ -0,0 +1,415 @@
+       if (poi.userIsWithin) {
+           // the user is in the poi, now we need to make sure we haven't already recorded the entry event
+           if (![regionsUserIsCurrentlyWithin containsObject:poi.identifier]) {
+               CLCircularRegion *region = [[CLCircularRegion alloc] initWithCenter:CLLocationCoordinate2DMake(poi.latitude, poi.longitude)
+                                                                            radius:poi.radius
+                                                                        identifier:poi.identifier];
+               [ACPPlaces processRegionEvent:region forRegionEventType:ACPRegionEventTypeEntry];
+           }
+
+           // add the region to our updated list
+           [updatedRegionsUserIsCurrentlyWithin addObject:poi.identifier];
+       }
+   }
+
+   // update user defaults with the new list
+   [[NSUserDefaults standardUserDefaults] setObject:updatedRegionsUserIsCurrentlyWithin forKey:@"regionsUserIsAlreadyIn"];
+}
+```
+
+{% endtab %}
+
+{% tab title="iOS (Swift)" %}
+
+This code sample shows handling the result provided in the callback of `getNearbyPoints(_ ofInterest: CLLocation, limit: UInt, callback: (([ACPPlacesPoi]?) -> Void)?, errorCallback: ((ACPPlacesRequestError) -> Void)?)`, an `[ACPPlacesPoi]`:
+
+```swift
+func handleUpdatedPOIs(_ nearbyPois:[ACPPlacesPoi]) {
+    // get the list of regions we know the user is already within from user defaults
+    let regionsUserIsCurrentlyWithin : [String] = UserDefaults.standard.array(forKey: "regionsUserIsAlreadyIn") as! [String]
+
+    // loop through new nearbyPoi and post entry events for pois that aren't already in our list
+    // also creating the new list of known regions that the user is in
+    var updatedRegionsUserIsCurrentlyWithin: [String] = []
+    for poi in nearbyPois {
+        // check if the user is in this poi
+        if poi.userIsWithin {
+            // the user is in the poi, now we need to make sure we haven't already recorded the entry event
+            if !regionsUserIsCurrentlyWithin.contains(poi.identifier!) {
+                let region = CLCircularRegion.init(center: CLLocationCoordinate2D.init(latitude: poi.latitude, longitude: poi.longitude), radius: CLLocationDistance(poi.radius), identifier: poi.identifier!)
+                ACPPlaces.processRegionEvent(region, for: .entry)
+            }
+
+            // add the region to our updated list
+            updatedRegionsUserIsCurrentlyWithin.append(poi.identifier!)
+        }
+    }
+
+    // update user defaults with the new list
+    UserDefaults.standard.set(updatedRegionsUserIsCurrentlyWithin, forKey: "regionsUserIsAlreadyIn")
+}
+```
+
+{% endtab %}
+
+{% endtabs %}
+
+## Putting the Pieces Together
+
+The following code samples should help put the picture together. They show retrieving the current location of the device, triggering the necessary events, and ensuring that you don't get multiple entries for the same location on a single visit.
```

Comment:
- Please change the title to "Sample implementation"...would this work?

- Please reword to:
"The following code samples show how to retrieve the current location of the device, trigger the necessary events, and ensure that you do not get multiple entries for the same location on one visit: